google/cloud/storage/examples/storage_bucket_samples.cc

```diff
@@ -383,6 +383,96 @@ void RemoveBucketDefaultKmsKey(google::cloud::storage::Client client, int& argc,
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [enable bucket policy only]
+  // [START storage_enable_bucket_policy_only]
+  namespace gcs = google::cloud::storage;
+  using google::cloud::StatusOr;
+  [](gcs::Client client, std::string bucket_name) {
+    gcs::BucketIamConfiguration configuration;
+    configuration.bucket_policy_only = gcs::BucketPolicyOnly{true};
+    StatusOr<gcs::BucketMetadata> updated_metadata = client.PatchBucket(
+        bucket_name, gcs::BucketMetadataPatchBuilder().SetIamConfiguration(
+                         std::move(configuration)));
+
+    if (!updated_metadata) {
+      throw std::runtime_error(updated_metadata.status().message());
+    }
+
+    std::cout << "Successfully enabled Bucket Policy Only on bucket "
+              << updated_metadata->name() << "\n";
+  }
+  // [END storage_enable_bucket_policy_only]
+  //! [enable bucket policy only]
+  (std::move(client), bucket_name);
+}
+
+void DisableBucketPolicyOnly(google::cloud::storage::Client client, int& argc,
+                             char* argv[]) {
+  if (argc != 2) {
+    throw Usage{"disable-bucket-policy-only <bucket-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [disable bucket policy only]
+  // [START storage_disable_bucket_policy_only]
+  namespace gcs = google::cloud::storage;
+  using google::cloud::StatusOr;
+  [](gcs::Client client, std::string bucket_name) {
+    gcs::BucketIamConfiguration configuration;
+    configuration.bucket_policy_only = gcs::BucketPolicyOnly{false};
+    StatusOr<gcs::BucketMetadata> updated_metadata = client.PatchBucket(
+        bucket_name, gcs::BucketMetadataPatchBuilder().SetIamConfiguration(
+                         std::move(configuration)));
+
+    if (!updated_metadata) {
+      throw std::runtime_error(updated_metadata.status().message());
+    }
+
+    std::cout << "Successfully disabled Bucket Policy Only on bucket "
+              << updated_metadata->name() << "\n";
+  }
+  // [END storage_disable_bucket_policy_only]
+  //! [disable bucket policy only]
+  (std::move(client), bucket_name);
+}
+
+void GetBucketPolicyOnly(google::cloud::storage::Client client, int& argc,
+                         char* argv[]) {
+  if (argc != 2) {
+    throw Usage{"get-bucket-policy-only <bucket-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [get bucket policy only]
+  // [START storage_get_bucket_policy_only]
+  namespace gcs = google::cloud::storage;
+  using google::cloud::StatusOr;
+  [](gcs::Client client, std::string bucket_name) {
+    StatusOr<gcs::BucketMetadata> bucket_metadata = client.GetBucketMetadata(bucket_name);
+
+    if (!bucket_metadata) {
+      throw std::runtime_error(bucket_metadata.status().message());
+    }
+
+    if (bucket_metadata->has_iam_configuration()) {
+      google::cloud::optional<gcs::BucketPolicyOnly> bucket_policy_only = bucket_metadata->iam_configuration().bucket_policy_only;
```

Comment:
Please dereference the optional instead to get `gcs::BucketPolicyOnly` instead of surfacing `google::cloud::optional`.

Reply:
OK done. But should I also check if `has_value()`?

Reply:
That sounds like a good idea, thanks for calling that out. Yes add `has_value()`.